### PR TITLE
Fix breaking change in upgrading OTEL Collector

### LINKED ISSUE #5302: Fix breaking changes in upgrading OTEL Collector
PR #5301. Looks like breaking changes in the collector.

One of the errors: `Error: ../../../go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/receiver/zipkinreceiver@v0.96.0/trace_receiver.go:61:25: undefined: component.ErrNilNextConsumer`

### PR DESCRIPTION
**Pull request by lekaf974**
## Which problem is this PR solving?
- Fix #5302 

## Description of the changes
- 

## How was this change tested?
- 

## Checklist
- [X] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [X] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -158,17 +158,17 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/syslogexporter v0.96.0 // indirect
 	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/zipkinexporter v0.96.0 // indirect
 	github.com/open-telemetry/opentelemetry-collector-contrib/extension/storage v0.96.0
-	github.com/open-telemetry/opentelemetry-collector-contrib/internal/common v0.96.0 // indirect
-	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.96.0 // indirect
-	github.com/open-telemetry/opentelemetry-collector-contrib/internal/kafka v0.96.0 // indirect
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/common v0.97.0 // indirect
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.97.0 // indirect
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/kafka v0.97.0 // indirect
 	github.com/open-telemetry/opentelemetry-collector-contrib/internal/sharedcomponent v0.96.0 // indirect
```
Comment (yurishkuro) [inline go.mod:164]: we don't upgrade piecemeal, it should include all OTEL packages

Comment (lekaf974) [reply go.mod]: @yurishkuro think I have update all OTEL packages, ready for another review 

PS: I saw this warning in my IDE
image

**Comment by yurishkuro** [comment]
please make sure `make test` succeeds locally before submitting for review

**Comment by lekaf974** [comment]
> please make sure `make test` succeeds locally before submitting for review

`make test` is passing locally

Ready for a new review

File: cmd/jaeger/internal/components_test.go [hunk 2]
```diff
@@ -38,9 +38,6 @@ func TestComponents(t *testing.T) {
 	assert.NotNil(t, factories.Exporters)
 	assert.NotNil(t, factories.Processors)
 	assert.NotNil(t, factories.Connectors)
-
-	_, jaegerReceiverFactoryExists := factories.Receivers["jaeger"]
-	assert.True(t, jaegerReceiverFactoryExists)
```
Comment (yurishkuro) [inline cmd/jaeger/internal/components_test.go:43]: Why is this deleted?

Comment (lekaf974) [reply cmd/jaeger/internal/components_test.go:43]: good catch did remove it temporary while I was fixing the tests and forgot to add it in the last commit whit the new implementation. 

Done

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
@lekaf974 thanks for the help!
